include/field3d.hxx

```diff
@@ -295,6 +295,13 @@ class Field3D : public Field, public FieldData {
     return data[d.ind];
   }
 
+  BoutReal& operator[](const Indices &d) {
```

Comment:
I'd like to avoid bringing these operators back if possible. If we added something like

```cpp
// const Mesh& would be better, but I think Mesh const-ness might need fixing
Ind3D indexFromXYZ(int x, int y, int z, Mesh* mesh)
```

would that avoid the need for `Indices`? If not, then I would suggest just moving `Indices` entirely into `boundary_nonuniform.cxx` and just making it POD:

```cpp
struct Indices {
  int x, y, z;
};
```

and explicitly indexing the fields with `f(i.x, i.y, i.z)`

Reply:
I like Indices because they allow better readability, and are also not worse then integer indices performance wise. Further they allow to write code for Fields. Finally reintroducing them removes a breaking change from the 4.2(?) minor release.

I did not understand why they were removed in the first place, could you explain?

Reply:
We removed them because we don't want to have to maintain many different ways of indexing fields. Also `Indices` was really an implementation detail of `DataIterator`. Now that we have improved on `DataIterator` it doesn't make sense to keep `Indices` about as well.